New lightweight bss theme changes, to be finalise testing by ranisa

ranisa-gupta16 opened the pull request:
# feat/fix/chore: pr title

## JIRA Ticket

[[<JIRA_TICKET>](https://jira.csiro.com/browse/<JIRA_TICKET>)
](https://jira.csiro.au/browse/BSS-436)


SCREENSHOTS:

![image](https://github.com/user-attachments/assets/b57148e9-25e2-4cd3-9477-4cd5589e01ba)

![image](https://github.com/user-attachments/assets/543f3216-67f6-40b3-baae-9f02f2a12958)

![image](https://github.com/user-attachments/assets/7a4e0004-dfb8-4105-9e59-7589b34a8518)

![image](https://github.com/user-attachments/assets/5ed97df2-b460-46ad-93b6-0216c06ff734)

![image](https://github.com/user-attachments/assets/9130e8f7-a67e-440b-beda-543abb07146d)

![image](https://github.com/user-attachments/assets/97dd5db2-1dbf-451a-8417-610e7f530fee)

more.....


Add any additional information or context about the pull request here.

## Checklist

- [y ] I have confirmed all commits have been signed.
- [y ] I have added JSDoc style comments to any new functions or classes.
- [y ] Relevant documentation such as READMEs, guides, and class comments are updated.

PeterBaker0 commented:
On desktop I see this for the profile icon?

![image](https://github.com/user-attachments/assets/17ce317d-03dc-4aa0-bdeb-654963c61a35)

PeterBaker0 commented:
![image](https://github.com/user-attachments/assets/9469c151-fc70-4fe4-98b4-339a61a0e867)
Were these small arrows always here? On the right?

PeterBaker0 commented:
I think the bubble theme is broken - namely in sync_switch it expects a `theme.palette.icon.main` which is undefined for bubble. 
sync_switch.tsx
```
<Box>
          <FormControlLabel
            sx={{mr: 0, md: 2}}
            control={
              <Switch
                checked={project.sync}
                disabled={false}
                onClick={handleOpen}
                sx={{
                  '& .MuiSwitch-switchBase.Mui-checked': {
                    **color: theme.palette.icon.main,**
                  },
```

PeterBaker0 reviewed (changes requested):
My profile icon shows as a white background, and bubble is throwing an error at the moment. Otherwise looks good!

ranisa-gupta16 commented on app/.env.dist:
```diff
@@ -15,8 +15,8 @@ VITE_DEBUG_APP=true
-# The theme to use for the app, 'default' or 'bubble'
-VITE_THEME='default'
+# The theme to use for the app, 'default' or 'bubble' or 'bssTheme'
+VITE_THEME='bssTheme'
```
Yes i'll set this to default, if we need bss we'll switch it.

ranisa-gupta16 replied:
Bubble theme is not working that was the reason of creating a new theme.
I believe we would need to either spend time fixing that theme in future or remove it.
I'll add a ticket to look into that in backlog.

ranisa-gupta16 commented on app/public/manifest.json:
Ah no, I need to uncommit this.

ranisa-gupta16 commented on app/src/gui/components/notebook/record_table.tsx:
yes I agree. It's the best case approach right now to have those theming in. We'll need to do some improvisations moving ahead.

ranisa-gupta16 commented on app/src/gui/components/notebook/settings/index.tsx:
```diff
@@ -111,6 +112,15 @@ export default function NotebookSettings(props: {uiSpec: ProjectUIModel}) {
+                        '& .MuiSwitch-switchBase.Mui-checked': {
+                          color: theme.palette.icon.main,
+                        },
+                        '& .MuiSwitch-switchBase.Mui-checked + .MuiSwitch-track':
```
I find it from chrome dev tools, 

whatever section I'm looking at , I get to its root 
![image](https://github.com/user-attachments/assets/7b1cf081-5fa4-4c48-b963-223b20c6adf6)
![image](https://github.com/user-attachments/assets/21fa0ffa-88aa-4daf-b675-59483558efbc)

and if I need more info I refer to Mui CSS Styles API - here's the link https://mui.com/system/styles/api/

ranisa-gupta16 commented on app/src/gui/themes/bssTheme/appBar.tsx:
```diff
@@ -0,0 +1,69 @@
+
+const drawerWidth = 240;
+
+// bss theme related elements to be tested  - @todo RAnisa
```
This is done, updating this todo.
Updating this todo.

ranisa-gupta16 commented:
> ![image](https://private-user-images.githubusercontent.com/87056634/381320332-9469c151-fc70-4fe4-98b4-339a61a0e867.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzAyNDE0NTksIm5iZiI6MTczMDI0MTE1OSwicGF0aCI6Ii84NzA1NjYzNC8zODEzMjAzMzItOTQ2OWMxNTEtZmM3MC00ZmU0LTk4YjQtMzM5YTYxYTBlODY3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDI5VDIyMzIzOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTE0ODI2NWI1ZGQxMTBjNWFiZGNkNjhhNDQ2MWJhZGM3YWI0MDRhMTk5YTJiY2QxYjEzYzc2YjA0NjkwZjdlYTkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.9LrNEOG1pVGir9GXXZc-0Zekt5qmK962wSez4IHxiXQ) Were these small arrows always here? On the right?

Yes this is preexisting , might try to fix this

ranisa-gupta16 commented:
> On desktop I see this for the profile icon?
> 
> ![image](https://private-user-images.githubusercontent.com/87056634/381319163-17ce317d-03dc-4aa0-bdeb-654963c61a35.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzAyNDE0NTksIm5iZiI6MTczMDI0MTE1OSwicGF0aCI6Ii84NzA1NjYzNC8zODEzMTkxNjMtMTdjZTMxN2QtMDNkYy00YWEwLWJkZWItNjU0OTYzYzYxYTM1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDI5VDIyMzIzOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTU4N2E0ZWNmNjY5Mzc2NmQ2OWI3ZDc1ZDQ2OGViMWI3NzQ0MTI5N2YxNzc3YThkMTg5NDk5NTljYWRjNWJmMTgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.fLsZSqcHN489_SzU43tJaK4SAlNmQTvv8z2PGQgEpFw)

Ammm, I don;'t see this, may be hard refresh and see, I see sign in in both mobile and desktop versions for both the themes.
![image](https://github.com/user-attachments/assets/7d7dc963-4196-4571-9db6-a94bc652a98a)

![image](https://github.com/user-attachments/assets/d744127d-3989-4f5c-95a7-5ef13c556fef)

ranisa-gupta16 commented:
> I think the bubble theme is broken - namely in sync_switch it expects a `theme.palette.icon.main` which is undefined for bubble. sync_switch.tsx
> 
> ```
> <Box>
>           <FormControlLabel
>             sx={{mr: 0, md: 2}}
>             control={
>               <Switch
>                 checked={project.sync}
>                 disabled={false}
>                 onClick={handleOpen}
>                 sx={{
>                   '& .MuiSwitch-switchBase.Mui-checked': {
>                     **color: theme.palette.icon.main,**
>                   },
> ```
Bubble theme have had some issues. That was the reason we've created a new theme and not using bubble anymore.
We might have to fix bubble in future may be or that'll be removed down the line.

ranisa-gupta16 commented on app/src/gui/pages/record-create.tsx:
```diff
@@ -280,10 +280,16 @@ function DraftEdit(props: DraftEditProps) {
               textColor="secondary"
             >
               <Tab label="Create" value="1" sx={{color: '#c2c2c2'}} />
-              <Tab label="Meta" value="2" sx={{color: '#c2c2c2'}} />
+              <Tab label="Meta" value="2" sx={{color: '#C13E3EFF'}} />
```
Yup this was fixed based on theming, needed to remove these.

ranisa-gupta16 commented:
> > ![image](https://private-user-images.githubusercontent.com/87056634/381320332-9469c151-fc70-4fe4-98b4-339a61a0e867.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzAyNDE0NTksIm5iZiI6MTczMDI0MTE1OSwicGF0aCI6Ii84NzA1NjYzNC8zODEzMjAzMzItOTQ2OWMxNTEtZmM3MC00ZmU0LTk4YjQtMzM5YTYxYTBlODY3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDI5VDIyMzIzOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTE0ODI2NWI1ZGQxMTBjNWFiZGNkNjhhNDQ2MWJhZGM3YWI0MDRhMTk5YTJiY2QxYjEzYzc2YjA0NjkwZjdlYTkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.9LrNEOG1pVGir9GXXZc-0Zekt5qmK962wSez4IHxiXQ) Were these small arrows always here? On the right?
> 
> Yes this is preexisting , might try to fix this

Ahh I see, there was a unnecessary scroll added for this which is why it was displaying those arrows.
Fixed it.

ranisa-gupta16 commented:
@PeterBaker0 , I have fixed the issues, resolved all conversations for this PR and have also added comments on the ones that doesn't need fixing.
Please review this when you get a chance.  :-)

PeterBaker0 commented:
> > I think the bubble theme is broken - namely in sync_switch it expects a `theme.palette.icon.main` which is undefined for bubble. sync_switch.tsx
> > ```
> > <Box>
> >           <FormControlLabel
> >             sx={{mr: 0, md: 2}}
> >             control={
> >               <Switch
> >                 checked={project.sync}
> >                 disabled={false}
> >                 onClick={handleOpen}
> >                 sx={{
> >                   '& .MuiSwitch-switchBase.Mui-checked': {
> >                     **color: theme.palette.icon.main,**
> >                   },
> > ```
> 
> Bubble theme have had some issues. That was the reason we've created a new theme and not using bubble anymore. We might have to fix bubble in future may be or that'll be removed down the line.

as discussed, theme should either work, or be removed - it's okay if the colours are not good.

PeterBaker0 commented:
> @PeterBaker0 , I have fixed the issues, resolved all conversations for this PR and have also added comments on the ones that doesn't need fixing. Please review this when you get a chance. :-)

See fix I pushed for sign in styling - all of your screenshots show logged out

PeterBaker0 reviewed (approved):
Once bubble theme functions again - even if it looks bad, is fine with me. I have pushed a proposed fix for the sign in styling.
